src/pudl/transform/ferc1.py

```diff
@@ -535,6 +538,95 @@ def unstack_balances_to_report_year_instant_xbrl(
+    slightly different names (ex: ``residential_sales_axis`` vs.
+    ``industrial_sales_axis``), we'll need to combine them. We combine them to get rid
+    of NA values which aren't allowed in primary keys. Otherwise it would look like
+    this:
+
+        +-------------------------+-------------------------+
+        | residential_sales_axis  | industrial_sales_axis   |
+        +=========================+=========================+
+        | value1                  | NA                      |
+        +-------------------------+-------------------------+
+        | value2                  | NA                      |
+        +-------------------------+-------------------------+
+        | NA                      | valueA                  |
+        +-------------------------+-------------------------+
+        | NA                      | valueB                  |
+        +-------------------------+-------------------------+
+
+    vs. this:
+
+        +-------------------------+
+        | sales_axis              |
+        +=========================+
+        | value1                  |
+        +-------------------------+
+        | value2                  |
+        +-------------------------+
+        | valueA                  |
+        +-------------------------+
+        | valueB                  |
+        +-------------------------+
+    """
+    # First, make sure that the new_axis_column_name param as the word axis in it
+    if not params.new_axis_column_name.endswith("_axis"):
+        raise ValueError(
+            "Your new_axis_column_name must end with the suffix _axis so that it gets "
+            "included in the record_id."
+        )
+    # Now, make sure there are no overlapping axis columns
+    if (df[params.axis_columns_to_combine].count(axis=1) > 1).any():
+        raise AssertionError(
+            "You're trying to combine axis columns, but there's more than one axis "
+            "column value per row."
+        )
+    # Now combine all of the columns into one and remove the old axis columns
+    df[params.new_axis_column_name] = pd.NA
+    for col in params.axis_columns_to_combine:
+        df[params.new_axis_column_name] = df[params.new_axis_column_name].fillna(
+            df[col]
+        )
+    df = df.drop(columns=params.axis_columns_to_combine)
+
+    # This is bad, but somehow there are NA values in the sales_by_rate_schedule table
+    # axis columns, which this function is basically made for. Soooo in order for it not
+    # to break when it uses the axis columns as pk values, just fill them with blanks
+    # here. Yike.
+    if df[params.new_axis_column_name].isna().any():
+        logger.warning(
+            "Some of the newly combined axis column values are NA...we're blanking "
+            "them out now, but that's not ideal!"
+        )
+        df[params.new_axis_column_name] = df[params.new_axis_column_name].fillna("")
```

Comment:
hm yeah this is a bummer! totally see why you did it though :-/ ... If this is a `sales_by_rate_schedule` problem I wonder if we can fix it in a `sales_by_rate_schedule` specific way and possibly closer to where this would actually cause a problem, like possibly overriding the `assign_record_id` with a quick fill-in na's and then run the default method? but also i'm not sure where else this would be a problem so I'm assuming its a `assign_record_id` issue.